lib/real_world/http/mappers/retriever.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'rack'
+
+module RealWorld
+  module Http
+    module Mappers
+      class Retriever
+        attr_reader :retrievers
+
+        def initialize(*retrievers)
+          @retrievers = retrievers
+        end
+
+        def call(param_definitions, env)
+          params = {}
+          request = Rack::Request.new(env)
+          retrievers = @retrievers.map { |retriever| retriever.new(request, env) }
+
+          param_definitions.each do |name:, required:|
+            name = name.to_sym
+            found = false
+
+            retrievers.each do |retriever|
+              value, found, error = retriever.call(name)
+
+              return nil, error if error
+
+              if found
+                params[name] = value
+                break
+              end
+            end
+
+            return nil, { type: :param_not_found, data: { name: name } } if required && !found
```

Comment:
Note: Include the `Colmena::Error` mixin and use the `error` method to make it explicit that `:param_not_found` in an error with the conventional structure

Reply:
Done in my next PR